assets/src/dashboard/components/detailViewNavBar/index.js

```diff
@@ -79,23 +79,32 @@ const CapitalizedButton = styled(Button)`
   text-transform: uppercase;
 `;
 
-export function TemplateNavBar({ handleCta, handleBookmarkClick }) {
+export function DetailViewNavBar({
+  handleCta,
+  handleBookmarkClick,
+  isBookmarkingEnabled,
+  ctaText,
+}) {
   return (
     <Nav>
       <Container>
         <CloseLink href={parentRoute()}>{__('Close', 'web-stories')}</CloseLink>
       </Container>
       <Container>
-        {handleBookmarkClick && <BookmarkToggle />}
+        {isBookmarkingEnabled && (
+          <BookmarkToggle onClick={handleBookmarkClick} />
```

Comment:
This is fine, but if this `isBookmarkingEnabled` is true and they do not pass a `handleBookmarkClick` we will get a prop exception. An alternative here is to use the existence of a `handleBookmarkClick` to determine if bookmarking is enabled or not. Without the handler, the functionality isn't useful anyway.

Reply:
I like @BrittanyIRL's solution, would making sure that `handleBookmarkClick` defaults to `() => {}` be good middle ground?

Reply:
Actually on second thought that may lead to weird behavior (click and nothing happens), I think checking the existence of both the flag and that the callback is truthy is probably best.

Reply:
sure, that makes sense!

Reply:
@dmmulroy I'm curious, it seems that at that point `isBookmarkingEnabled` doesn't add any value. The only reason this is there at all right now is the api for bookmarking doesn't exist and the functionality isn't scoped out yet. I think the simplest way forward is to just not pass in `handleBookmarkClick`, check for it to be defined before rendering said bookmark button and then in the future once bookmarking is ready to get added this is just a natural progression of the component not so much 'flagging' but adding functionality.

Reply:
If you're suggesting lifting the `isBookmarkingEnabled` to the parent and using it to conditionally pass the callback to the child I think that makes sense.  I don't think we want to get rid of the flag since we want the wordpress/php file to control whether it's enabled or not.

Reply:
that would mean that `handleBookmarkClick`'s type will change - either as a callback or undefined since we don't want it to be defined unless it's in use - which i think was part of the original conversation about this lil component 😂  - are you cool with `handleBookmarkClick` getting passed as undefined when the bookmarking enabled flag is false?

Reply:
thought about this a little more, see if the update i made is alright? i think it's inline with what you're saying.